### ISSUE TITLE
array_key_exists(): Argument #2 ($array) must be of type array, Stripe\Util\CaseInsensitiveArray given

### ISSUE THREAD

**Issue by mandeepsinghn** [stripe/stripe-php#1125]
TypeError
File:
/vendor/stripe/stripe-php/lib/ApiRequestor.php:402
Message:
array_key_exists(): Argument #2 ($array) must be of type array, Stripe\Util\CaseInsensitiveArray given

This error exists in Stripe library. Please fix it asap. PHP version is 8.0.5

**Comment by paulasjes-stripe** [comment]
Can you provide more details? Like some sample code that recreates the error or a stack trace?

**Comment by mandeepsinghn** [comment]
It is not related to any code sample. Please use PHP 8 for testing and when you are going to add a card for the customer you are going to get this error. This is type related issue because in PHP 8 array_key_exist is not working on non array times and here  In ApiRequestor.php we are getting object instead of array. For php7 and lower version it is working fine.

**Comment by remi-stripe** [comment]
@mandeepsinghn Which version of stripe-php are you using currently? Your callstack says you crash on line 402 of API Requestor but on the latest major [this line](https://github.com/stripe/stripe-php/blob/master/lib/ApiRequestor.php#L402) does not use `array_key_exists`. Is it possible that you're on an old version of our library from before we fixed this?

**Comment by mandeepsinghn** [comment]
We are using the latest version of stripe-php but the line number is different because we tried to debug it. The original line number is 370. We also found a solution for this. I am sharing the code with you. We fixed it in our project but we also want to share this problem & solution with you because PHP8 is the future.
/////////
if(is_array($params))return \array_key_exists($key, $params);
if(is_object($params))return \property_exists($params,$key);

**Comment by remi-stripe** [comment]
Perfect, thanks for providing the code change, we'll be able to look into fixing this and adding a test to catch more of those for PHP8

**Comment by richardm-stripe** [comment]
Hello @mandeepsinghn,

I'm curious if you can provide more information about how this error was triggered.

I cannot figure out how it would have been possible trigger the error you reported. The line throwing the error
```php
return \array_key_exists($key, $params);
```
is wrapped in an if statement  (and [has been](https://github.com/stripe/stripe-php/commit/429cfca95cd49dc322c410dc608ad654c7184483) since it was introduced)
```php
if ($params && \is_array($params)) {
```
and `\array_key_exists($params)` should never throw an error when `\is_array($params)` returns false. The only way I can trigger the error is if I remove the `is_array($params)` check.

**Comment by nick-kanel** [comment]
@richardm-stripe I just ran into this exact error on php 8.0.13 with stripe-php version 6.43.1. Upgrading stripe-php to 7.105.0 seems to fix this error.

**Comment by rwellens** [comment]
Can you still fix it in version 6? There are customers who switch to PHP8 without knowing that there is a huge BC break in their payment system

**Comment by Jattura** [comment]
Yes please fix it as many old magento 1.9 have been upgraded to php 8 or 8.1. Best, please help with a new patch, thanks.

**Comment by kevinjavitz** [comment]
This is what worked for me, 

Change lib/Stripe/lib/ApiRequestor.php:403

`if (array_key_exists('request-id', $rheaders)) {`

to:

`if (\property_exists($rheaders,'request-id')) {`
